internal/services/synapse/synapse_managed_private_endpoint_resource.go

```diff
@@ -112,16 +114,29 @@ func resourceSynapseManagedPrivateEndpointCreate(d *pluginsdk.ResourceData, meta
 			GroupID:               utils.String(d.Get("subresource_name").(string)),
 		},
 	}
-	resp, err := client.Create(ctx, virtualNetworkName, privateEndpointName, managedPrivateEndpoint)
+	resp, err := client.Create(ctx, id.ManagedVirtualNetworkName, id.Name, managedPrivateEndpoint)
 	if err != nil {
-		return fmt.Errorf("creating Synapse Managed Private Endpoint %q (Workspace %q / Resource Group %q): %+v", privateEndpointName, workspaceId.Name, workspaceId.ResourceGroup, err)
+		return fmt.Errorf("creating %s: %+v", id, err)
 	}
 
 	if resp.ID == nil || *resp.ID == "" {
-		return fmt.Errorf("empty or nil ID returned for Synapse Managed Private Endpoint %q (Workspace %q / Resource Group %q)", privateEndpointName, workspaceId.Name, workspaceId.ResourceGroup)
+		return fmt.Errorf("empty or nil ID returned for %s", id)
+	}
+
+	timeout, _ := ctx.Deadline()
+
+	stateConf := &pluginsdk.StateChangeConf{
+		Pending:    []string{string(synapse.ProvisioningStateProvisioning)},
+		Target:     []string{string(synapse.ProvisioningStateSucceeded)},
+		Refresh:    managedPrivateEndpointProvisioningStateRefreshFunc(ctx, client, id),
+		MinTimeout: 1 * time.Minute,
+		Timeout:    time.Until(timeout),
```

Comment:
it'd be worth setting an explicit interval for this, since I think the default is 5s:

```suggestion
		Timeout:    time.Until(timeout),
		Interval: 15 * time.Second,
```